server/sonar-server/src/main/java/org/sonar/server/es/RecoveryIndexer.java

```diff
@@ -65,20 +65,16 @@
   private final System2 system2;
   private final Configuration config;
   private final DbClient dbClient;
-  private final UserIndexer userIndexer;
-  private final RuleIndexer ruleIndexer;
-  private final ActiveRuleIndexer activeRuleIndexer;
+  private final Map<IndexType, ResilientIndexer> indexersByType;
   private final long minAgeInMs;
   private final long loopLimit;
 
-  public RecoveryIndexer(System2 system2, Configuration config, DbClient dbClient,
-    UserIndexer userIndexer, RuleIndexer ruleIndexer, ActiveRuleIndexer activeRuleIndexer) {
+  public RecoveryIndexer(System2 system2, Configuration config, DbClient dbClient, ResilientIndexer... indexers) {
     this.system2 = system2;
     this.config = config;
     this.dbClient = dbClient;
-    this.userIndexer = userIndexer;
-    this.ruleIndexer = ruleIndexer;
-    this.activeRuleIndexer = activeRuleIndexer;
+    this.indexersByType = new HashMap<>();
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Implement "Comparable" in "IndexType" or change the key type of this map. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS3552)